### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by roman-khimov**
Fix some of the old bugs related to this.

### CODE REVIEW

File: docs/compiler.md [hunk 1]
```diff
@@ -33,6 +33,13 @@ a dialect of Go rather than a complete port of the language:
    it's up to the programmer whether assert can be performed successfully.
```
Comment (AliceInHunterland) [inline docs/compiler.md:33]: Format of the commit message is questionable. @AnnaShaleva would not like that.

Comment (roman-khimov) [reply docs/compiler.md:33]: Do you think it violates https://github.com/nspcc-dev/.github/blob/master/git.md#commit-messages?

Comment (AliceInHunterland) [reply docs/compiler.md:33]: yes,

> The following format is expected:
> 
> subsystem: short description (up to 80 symbols, better up to 60)
> 
> Long description with references, links...

in the first line there are no links or refs. At least they are not mentioned in https://github.com/nspcc-dev/.github/blob/master/git.md#commit-messages

Comment (roman-khimov) [reply docs/compiler.md:33]: While I think it's compliant (there are no limits of what short description can or can not include as long as it's short) and there are numerous commits like f4b61b5a7, b1bb12df4, d2a716221, eade327b9 and others, I can at the same time say that @AnnaShaleva has _never_ wrote a short description like that in ~1715 commits that I can see in this repo (~116 commits in neo-bench are the same). I think it's still OK, but maybe we can have our policies improved one way or another.

File: docs/node-configuration.md [hunk 2]
```diff
@@ -403,3 +403,25 @@ where:
 
   Note that `Transaction` is a NeoGo extension that isn't supported by the NeoC#
   node and must be disabled on the public Neo N3 networks.
+
+## DB compatibility
```
Comment (AliceInHunterland) [inline docs/node-configuration.md:407]: Needs the same indent as above.

**Review by AliceInHunterland — APPROVED** [review]

File: docs/node-configuration.md (hunk 2 above)
Comment (roman-khimov) [reply docs/node-configuration.md:407]: We have a list above and this is a new section, but I've improved the list a bit anyway.

File: docs/compiler.md (hunk 1 above)
Comment (AnnaShaleva) [reply docs/compiler.md:33]: > in the first line there are no links or refs. 

I'm OK with the commit formatting presented in this PR, links are allowed both in the commit header and in the body.

**Review by AnnaShaleva — COMMENTED** [review]
LGTM except a tiny #3545.

**Pull request merged**